core/changestream/change.go

```diff
@@ -26,3 +26,16 @@ type ChangeEvent interface {
 	// ChangedUUID returns the entity UUID of the change.
 	ChangedUUID() string
 }
+
+// Term represents a set of changes that are bounded by a coalesced set.
+// The notion of a term are a set of changes that can be run one at a time
+// asynchronously. Allowing changes within a given term to be signaled of a
+// change independently from one another.
+// Once a change within a term has been completed, only at that point
+// is another change processed, until all changes are exhausted.
+type Term interface {
```

Comment:
I wonder if `Term` is the best name for this? It doesn't really convey the intent.
Did we consider `ChangeSet` for example?

Reply:
Yes, although I don't mind if we switch away from `Term`, just something other than `ChangeSet`. `Term` was chosen for two reasons:

 1. Term refers to a single watermark of changes that can only be actioned once.
 2. A term is marked complete via its `Done` function.

If we name it `ChangeSet` it goes against point 2. It's no longer just a set of changes.